src/Common/src/System/IO/TempFileCollection.cs

```diff
@@ -5,11 +5,22 @@
 using System.Collections;
 using System.IO;
 
+// We keep this class in Common to allow utilizing it without taking a dependency on System.CodeDom
+#if CODEDOM
 namespace System.CodeDom.Compiler
+#else
+namespace System.IO.Internal
+#endif
```

Comment:
Do we need to change the namespace?  I'm wondering if just changing the visibility of the type is sufficient.